crates/ra_editor/src/scope/fn_scope.rs

```diff
@@ -88,7 +88,7 @@ pub struct ScopeEntry {
 }
 
 impl ScopeEntry {
-    fn new(pat: ast::BindPat) -> Option<ScopeEntry> {
+    fn new_opt(pat: ast::BindPat) -> Option<ScopeEntry> {
```

Comment:
Let's use `new` to name such methods? It's what stdlib does, so it looks like a bug in clippy:

https://doc.rust-lang.org/std/ffi/struct.CString.html#method.new

Reply:
Looks like this will be fixed with https://github.com/rust-lang-nursery/rust-clippy/pull/3338